src/net.cpp

```diff
@@ -146,8 +146,10 @@ uint16_t GetListenPort()
     // If -bind= is provided with ":port" part, use that (first one if multiple are provided).
     for (const std::string& bind_arg : gArgs.GetArgs("-bind")) {
         constexpr uint16_t dummy_port = 0;
+        // maybe bind_arg has "=onion"
+        const std::string truncated_bind_arg = bind_arg.substr(0, bind_arg.rfind('='));
 
-        const std::optional<CService> bind_addr{Lookup(bind_arg, dummy_port, /*fAllowLookup=*/false)};
+        const std::optional<CService> bind_addr{Lookup(truncated_bind_arg, dummy_port, /*fAllowLookup=*/false)};
         if (bind_addr.has_value() && bind_addr->GetPort() != dummy_port) return bind_addr->GetPort();
     }
```

Comment:
I think this change to `GetListenPort()` will produce unwanted results if the configuration goes like:
`-bind=1.1.1.1:1111=onion -bind=2.2.2.2:2222`. In that case, I guess it should return `2222` like before this change, but after it, it would return `1111`.